draft-irtf-cfrg-opaque.md

```diff
@@ -654,12 +657,10 @@ is that which is associated with the OPAQUE configuration (see {{configurations}
 
 See {{export-usage}} for details about the output export_key usage.
 
-#### StoreUserRecord
-
-The StoreUserRecord function stores the tuple `(envU, pkS, skS, pkU, kU)`,
-where envU and pkU are obtained from the input RegistrationUpload message in
-a record associated with the user's account idU. If `skS` and `pkS` are used for
-multiple users, the server can store these values separately and omit them from
+Upon completion of this function, the client MUST send its public key `pkU`, optional
+identity `idU` (depending on the mode), and corresponding envelope `envU` to the server,
+where they are then store as a tuple `(envU, pkU, kU, pkS, skS)`. If `skS` and `pkS` are
+used for multiple users, the server can store these values separately and omit them from
```

Comment:
Why doesn't the client just send an instance of the `RegistrationUpload` function, which includes `pkU` and `envU`? It doesn't need to send `idU` since that is taken care of in the application layer and not necessary for the OPAQUE protocol